tools/code-generation/smithy/codegen/cpp-smoke-tests-codegen/src/main/java/com/amazonaws/util/awsclientsmithygenerator/generators/SmithyC2JNamespaceMap.java

```diff
@@ -17,9 +17,16 @@ public class SmithyC2JNamespaceMap {
 
     private SmithyC2JNamespaceMap(String jsonString){
 
-        Type mapType = new TypeToken<Map<String, String>>(){}.getType();
-        Gson gson = new Gson();
-        this.SMITHY_C2J_SERVICE_NAME_MAP = gson.fromJson(jsonString, mapType);
+        if(jsonString.isEmpty())
```

Comment:
This is a fix when no mapping is provided

Reply:
when would there be no mapping file, dont we always pass it one? shouldn't this be an exception?

Reply:
This should work with or without mapping. Mapping is just an override. Codegen script passes mapping but it doesnt stop anyone from just using gradle build without one for testing (where no service needs to be remapped) will cause the issue

Reply:
yeah thats fair